lib/logstash/inputs/lumberjack.rb

```diff
@@ -29,19 +29,37 @@ class LogStash::Inputs::Lumberjack < LogStash::Inputs::Base
   # SSL key passphrase to use.
   config :ssl_key_passphrase, :validate => :password
 
+  # The lumberjack input using a fixed thread pool to do the actual work and
+  # will accept a number of client in a queue, before starting to refuse new
+  # connection. This solve an issue when logstash-forwarder clients are 
+  # trying to connect to logstash which have a blocked pipeline and will 
+  # make logstash crash with an out of memory exception.
+  config :max_clients, :validate => :number, :default => 1000
```

Comment:
Do you have an idea, to define the default value for it?
Do we have some data we could use to give it a sane default?